File: Python/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:

        stack = []
        dictionary = {
            ')' : '(',
            ']' : '[',
            '}' : '{'
        }   

        for char in s:

            if char in dictionary:
                if not stack:
                    return False
                top = stack.pop()
                if(top!=dictionary[char]):
                    return False
            else:
                stack.append(char) 

        return len(stack) == 0   

File: Javascript/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {

    const stack = []
    const map = {
        ')': '(',
        ']': '[',
        '}': '{'
    }

    for(char of s){

        if(char in map){
            let top = stack.pop()
            if(top !== map[char]){
                return false
            }

                }
                else{
                    stack.push(char)
                  
                }
    }

    return stack.length === 0
};